src/AlternateLife.RageMP.Net/Scripting/ScriptingClasses/Commands.cs

```diff
@@ -1,157 +1,207 @@
+            {typeof(ulong),   new PrimitiveParser<ulong>(ulong.TryParse)},
+            {typeof(string),  new StringParser()},
+            {typeof(Enum),    new EnumParser()}
+        };
+
+        private IEnumerable<object> ProcessArguments(IReadOnlyList<string> args, IReadOnlyList<ParameterInfo> expectedParameters)
+        {
+            if (args.Count < expectedParameters.Count)
+            {
+                return null;
+            }
+            var parsedArgs = new List<object>();
+            for (int i = 0; i < args.Count; i++)
             {
-                if (predicate(command.Value) == false)
+                var expectedType = expectedParameters[i].ParameterType;
+                var targetType = expectedType;
+                if (expectedType.BaseType == typeof(Enum))
                 {
-                    continue;
+                    expectedType = typeof(Enum);
                 }
+                bool isParams = IsParams(expectedParameters[i]);
 
-                _commands.TryRemove(command.Key, out _);
+                if (i == expectedParameters.Count - 1 && isParams)
+                {
+                    var paramList = new List<object>();
+                    for (; i < args.Count; i++)
+                    {
+                        paramList.Add(args[i]);
+                    }
+
+                    parsedArgs.Add(paramList.ToArray());
+                    break;
+                }
+
+                if (i == expectedParameters.Count - 1 && !isParams && args.Count > expectedParameters.Count)
+                {
+                    return null;
+                }
+                
+                if (!_typeParsingSwitch.ContainsKey(expectedType))
+                {
+                    return null;
+                }
+                
+                if (!_typeParsingSwitch[expectedType].TryParse(args[i], targetType, out var parsedPara))
+                {
+                    return null;
+                }
+                parsedArgs.Add(parsedPara);
             }
+
+            return parsedArgs;
         }
 
-        public async Task ExecuteCommand(IPlayer player, string commandText)
+        public async Task ExecuteCommand(IPlayer player, string text)
         {
-            if (string.IsNullOrWhiteSpace(commandText))
+            string[] commandMessage = text.Split(' ', StringSplitOptions.RemoveEmptyEntries);
+            if (commandMessage.Length == 0)
             {
                 return;
             }
+            string commandname = commandMessage[0];
 
-            var parts = commandText.Trim().Split(' ');
-
-            var commandName = parts[0];
-
-            if (_commands.TryGetValue(commandName, out var commandInformation) == false)
+            var command = _commands.FirstOrDefault(c => c.Name == commandname);
+            if (command == null)
             {
+                await player.OutputChatBoxAsync("Command not found.");
```

Comment:
I don't like that forced command error message. Maybe some kind of callback should be called so the person, that uses this library can output it themself.